File: Var10JS/ChessboardMovement.js
// Функция isValidMove, которая проверяет, является ли передвижение по координатам (x, y)
// в пределах доски размера N x M, исходя из направления move ('L' - влево, 'R' - вправо, 'U' - вверх, 'D' - вниз).
function isValidMove(x, y, N, M, move) {
    if (move === 'L' && y > 1) {
        return true;
    } else if (move === 'R' && y < M) {
        return true;
    } else if (move === 'U' && x > 1) {
        return true;
    } else if (move === 'D' && x < N) {
        return true;
    }
    return false;
}

// Функция findStartPosition, которая находит стартовую позицию на доске размера N x M,
// с учетом строки команд s.
function findStartPosition(N, M, s) {
    for (let i = 1; i <= N; i++) {
        for (let j = 1; j <= M; j++) {
            let x = i, y = j;
            let valid = true;

            for (const move of s) {
                if (!isValidMove(x, y, N, M, move)) {
                    valid = false;
                    break;
                }

                if (move === 'L') y--;
                else if (move === 'R') y++;
                else if (move === 'U') x--;
                else if (move === 'D') x++;
            }

            if (valid) {
                return [i, j];
            }
        }
    }
    return [-1, -1];
}

// Основная функция программы.
function main() {
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    rl.question("Enter size (N M): ", function(sizeInput) {
        const dimensions = sizeInput.trim().split(" ").map(Number);
        const N = dimensions[0];
        const M = dimensions[1];

        rl.question("Enter string of commands: ", function(commands) {
            const startPos = findStartPosition(N, M, commands);

            if (startPos[0] === -1 && startPos[1] === -1) {
                console.log("It is impossible to complete all commands without leaving the board.");
            } else {
                console.log("Result:", startPos);
            }

            rl.close();
        });
    });
}

// Вызов основной функции программы.
main();

File: Var10Python/ChessboardMovement.py
# Функция isValidMove, которая проверяет, является ли передвижение по координатам (x, y)
# в пределах доски размера N x M, исходя из направления move ('L' - влево, 'R' - вправо, 'U' - вверх, 'D' - вниз).
def is_valid_move(x, y, N, M, move):
    if move == 'L' and y > 1:
        return True
    elif move == 'R' and y < M:
        return True
    elif move == 'U' and x > 1:
        return True
    elif move == 'D' and x < N:
        return True
    return False

# Функция findStartPosition, которая находит стартовую позицию на доске размера N x M,
# с учетом строки команд s.
def find_start_position(N, M, s):
    for i in range(1, N+1):
        for j in range(1, M+1):
            x, y = i, j
            valid = True

            for move in s:
                if not is_valid_move(x, y, N, M, move):
                    valid = False
                    break

                if move == 'L':
                    y -= 1
                elif move == 'R':
                    y += 1
                elif move == 'U':
                    x -= 1
                elif move == 'D':
                    x += 1

            if valid:
                return (i, j)
    return (-1, -1)

# Основная функция программы.
def main():
    # Запрос на ввод размеров доски.
    N, M = map(int, input("Enter size (N M): ").split())

    # Запрос на ввод строки команд.
    s = input("Enter string of commands: ")

    # Нахождение стартовой позиции на доске с учетом строки команд.
    start_pos = find_start_position(N, M, s)

    # Вывод результата: стартовой позиции или сообщения о невозможности выполнения всех команд.
    if start_pos == (-1, -1):
        print("It is impossible to complete all commands without leaving the board.")
    else:
        print("Result:", start_pos)

# Вызов основной функции программы.
if __name__ == "__main__":
    main()
